Assets/Documentation/man/man3/AnzeigenMarker.3
.TH "AnzeigenMarker" 3 "Version 0.1" "NPP VR Simulation" \" -*- nroff -*-
.ad l
.nh
.SH NAME
AnzeigenMarker \- This class implements graphical cues to highlight a display the player is supposed to manipulate and to mark the target value that the player is supposed to set\&.  

.SH SYNOPSIS
.br
.PP
.PP
Inherits MonoBehaviour\&.
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "float \fBmoveSpeed\fP = 3f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fImoveSpeed\fP sppecifies the speed of vertical movements
.RE
.PP
"
.ti -1c
.RI "float \fBmoveAmount\fP = 0\&.015f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fImoveAmount\fP specifies the amount of vertical movement
.RE
.PP
"
.ti -1c
.RI "float \fBfontSizeSpeed\fP = 2f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIfontSizeSpeed\fP specifies the speed of the font size change
.RE
.PP
"
.ti -1c
.RI "float \fBfontSizeAmount\fP = 0\&.1f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIfontSizeAmount\fP specifies the amount of font size change
.RE
.PP
"
.ti -1c
.RI "float \fBtargetNumber\fP = 1000"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fItargetNumber\fP is used to calculate the position of the arrow pointing to the target value
.RE
.PP
"
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBStart\fP ()"
.br
.RI "This method initialises an arrow pointing to the target value, a "!" marking the display and the control component (\fBAnzeigeSteuerung\fP) of the display\&. "
.ti -1c
.RI "void \fBUpdate\fP ()"
.br
.RI "This method updates the arrow's position marking the target value and the font size of the "!" marking the display\&. "
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "Transform \fBpfeilTransform\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIpfeilTransform\fP is a Transfom for animating the arrow pointing to the target value
.RE
.PP
"
.ti -1c
.RI "TextMeshPro \fBtextMeshPro\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fItextMeshPro\fP is a TextMeshPro used to mark the display by an "!" 
.RE
.PP
"
.ti -1c
.RI "float \fBinitialY\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIinitialY\fP specifies the initial y position of the arrow pointing to the target value
.RE
.PP
"
.ti -1c
.RI "float \fBinitialFontSize\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIinitialFontSize\fP specifies the initial font size set for the TextMeshPro
.RE
.PP
"
.ti -1c
.RI "float \fBxleft\fP = 0\&.0611f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIxleft\fP specifies the leftmost position of the arrow indicating the target value
.RE
.PP
"
.ti -1c
.RI "float \fBxright\fP = \-0\&.0734f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIxright\fP specifies the rightmost position of the arrow indicating the target value
.RE
.PP
"
.ti -1c
.RI "\fBAnzeigeSteuerung\fP \fBanzeigeSteuerung\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIanzeigeSteuerung\fP is a reference to an \fBAnzeigeSteuerung\fP used to get the upper limit of a display to calculate the target number
.RE
.PP
"
.ti -1c
.RI "\fBAnzeigeSteuerung5\fP \fBanzeigeSteuerung2\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIanzeigeSteuerung2\fP is a reference to an \fBAnzeigeSteuerung5\fP used to get the upper limit of a display to calculate the target number
.RE
.PP
"
.in -1c
.SH "Detailed Description"
.PP 
This class implements graphical cues to highlight a display the player is supposed to manipulate and to mark the target value that the player is supposed to set\&. 
.SH "Member Function Documentation"
.PP 
.SS "void AnzeigenMarker\&.Start ()\fR [private]\fP"

.PP
This method initialises an arrow pointing to the target value, a "!" marking the display and the control component (\fBAnzeigeSteuerung\fP) of the display\&. 
.SS "void AnzeigenMarker\&.Update ()\fR [private]\fP"

.PP
This method updates the arrow's position marking the target value and the font size of the "!" marking the display\&. 
.SH "Member Data Documentation"
.PP 
.SS "\fBAnzeigeSteuerung\fP AnzeigenMarker\&.anzeigeSteuerung\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIanzeigeSteuerung\fP is a reference to an \fBAnzeigeSteuerung\fP used to get the upper limit of a display to calculate the target number
.RE
.PP

.SS "\fBAnzeigeSteuerung5\fP AnzeigenMarker\&.anzeigeSteuerung2\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIanzeigeSteuerung2\fP is a reference to an \fBAnzeigeSteuerung5\fP used to get the upper limit of a display to calculate the target number
.RE
.PP

.SS "float AnzeigenMarker\&.fontSizeAmount = 0\&.1f"

.PP

.PP
\fBParameters\fP
.RS 4
\fIfontSizeAmount\fP specifies the amount of font size change
.RE
.PP

.SS "float AnzeigenMarker\&.fontSizeSpeed = 2f"

.PP

.PP
\fBParameters\fP
.RS 4
\fIfontSizeSpeed\fP specifies the speed of the font size change
.RE
.PP

.SS "float AnzeigenMarker\&.initialFontSize\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIinitialFontSize\fP specifies the initial font size set for the TextMeshPro
.RE
.PP

.SS "float AnzeigenMarker\&.initialY\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIinitialY\fP specifies the initial y position of the arrow pointing to the target value
.RE
.PP

.SS "float AnzeigenMarker\&.moveAmount = 0\&.015f"

.PP

.PP
\fBParameters\fP
.RS 4
\fImoveAmount\fP specifies the amount of vertical movement
.RE
.PP

.SS "float AnzeigenMarker\&.moveSpeed = 3f"

.PP

.PP
\fBParameters\fP
.RS 4
\fImoveSpeed\fP sppecifies the speed of vertical movements
.RE
.PP

.SS "Transform AnzeigenMarker\&.pfeilTransform\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIpfeilTransform\fP is a Transfom for animating the arrow pointing to the target value
.RE
.PP

.SS "float AnzeigenMarker\&.targetNumber = 1000"

.PP

.PP
\fBParameters\fP
.RS 4
\fItargetNumber\fP is used to calculate the position of the arrow pointing to the target value
.RE
.PP

.SS "TextMeshPro AnzeigenMarker\&.textMeshPro\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fItextMeshPro\fP is a TextMeshPro used to mark the display by an "!" 
.RE
.PP

.SS "float AnzeigenMarker\&.xleft = 0\&.0611f\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIxleft\fP specifies the leftmost position of the arrow indicating the target value
.RE
.PP

.SS "float AnzeigenMarker\&.xright = \-0\&.0734f\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIxright\fP specifies the rightmost position of the arrow indicating the target value
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for NPP VR Simulation from the source code\&.


Assets/Skripte/UI/AnzeigenMarker.cs
using UnityEngine;
using TMPro;

/// <summary>
/// This class implements graphical cues to highlight a display the player is supposed to manipulate and to mark the target value that the player is supposed to set. 
/// </summary>
public class AnzeigenMarker : MonoBehaviour
{
    /// <param name="moveSpeed"> sppecifies the speed of vertical movements</param>
    public float moveSpeed = 3f;            
    /// <param name="moveAmount"> specifies the amount of vertical movement</param>
    public float moveAmount = 0.015f;
    /// <param name="fontSizeSpeed"> specifies the speed of the font size change</param>
    public float fontSizeSpeed = 2f;        
    /// <param name="fontSizeAmount"> specifies the amount of font size change</param>
    public float fontSizeAmount = 0.1f;     
    /// <param name="targetNumber"> is used to calculate the position of the arrow pointing to the target value</param>
    public float targetNumber = 1000;
    /// <param name="pfeilTransform"> is a Transfom for animating the arrow pointing to the target value</param>
    private Transform pfeilTransform;
    /// <param name="textMeshPro"> is a TextMeshPro used to mark the display by an "!" </param>
    private TextMeshPro textMeshPro;
    /// <param name="initialY"> specifies the initial y position of the arrow pointing to the target value</param>
    private float initialY;
    /// <param name="initialFontSize"> specifies the initial font size set for the TextMeshPro</param>
    private float initialFontSize;
    /// <param name="xleft"> specifies the leftmost position of the arrow indicating the target value</param>
    private float xleft = 0.0611f;
    /// <param name="xright"> specifies the rightmost position of the arrow indicating the target value</param>
    private float xright = -0.0734f;
    /// <param name="anzeigeSteuerung">is a reference to an AnzeigeSteuerung used to get the upper limit of a display to calculate the target number</param>
    private AnzeigeSteuerung anzeigeSteuerung;
    /// <param name="anzeigeSteuerung2"> is a reference to an AnzeigeSteuerung5 used to get the upper limit of a display to calculate the target number</param>
    private AnzeigeSteuerung5 anzeigeSteuerung2;

    /// <summary>
    /// This method initialises an arrow pointing to the target value, a "!" marking the display and the control component (AnzeigeSteuerung) of the display.
    /// <summary>
    void Start()
    {
        // Find the child GameObject named "Pfeil"
        
        pfeilTransform = transform.Find("Pfeil");
        if (pfeilTransform != null)
        {
            initialY = pfeilTransform.localPosition.y; // Store the initial y position
        }
        else
        {
            Debug.LogError("Child GameObject 'Pfeil' not found.");
        }

        // Find the TextMeshPro component in the child GameObject named "AusrufeZeichen"

        Transform ausrufeZeichenTransform = transform.Find("AusrufeZeichen");
        if (ausrufeZeichenTransform != null)
        {
            textMeshPro = ausrufeZeichenTransform.GetComponent<TextMeshPro>();
            if (textMeshPro != null)
            {
                initialFontSize = textMeshPro.fontSize; // Store the initial font size
            }
            else
            {
                Debug.LogError("TextMeshPro component not found on 'AusrufeZeichen'.");
            }
        }
        else
        {
            Debug.LogError("Child GameObject 'AusrufeZeichen' not found.");
        }
        
        // Find the AnzeigeSteuerung component in the parent GameObject
        
        anzeigeSteuerung = GetComponentInParent<AnzeigeSteuerung>();
        if (anzeigeSteuerung == null)
        {
            anzeigeSteuerung2 = GetComponentInParent<AnzeigeSteuerung5>();
            if (anzeigeSteuerung2 == null)
            {
                Debug.LogError("AnzeigeSteuerung component not found in parent GameObject.");
            }
        }
        
    }

    /// <summary>
    /// This method updates the arrow's position marking the target value and the font size of the "!" marking the display.
    /// </summary>
    void Update()
    {
        if (pfeilTransform != null)
        {
            // Move the GameObject up and down on the y-axis smoothly
            float newY = initialY + Mathf.Sin(Time.time * moveSpeed) * moveAmount;
            pfeilTransform.localPosition = new Vector3(pfeilTransform.localPosition.x, newY - 0.0025f, pfeilTransform.localPosition.z);
        }

        if (textMeshPro != null)
        {
            // Change the font size smoothly
            float newFontSize = initialFontSize + Mathf.Sin(Time.time * fontSizeSpeed) * fontSizeAmount;
            textMeshPro.fontSize = newFontSize;
        }
        if (anzeigeSteuerung != null)
        {
            // Calculate the x-position based on the targetNumber
            float t = targetNumber / anzeigeSteuerung.end_Number;
            float newX = Mathf.Lerp(xleft, xright, t);
            // Update the x-position of the Pfeil
            pfeilTransform.localPosition = new Vector3(newX, pfeilTransform.localPosition.y, pfeilTransform.localPosition.z);
        }
        else if (anzeigeSteuerung2 != null)
        {
            // Calculate the x-position based on the targetNumber
            float t = targetNumber / anzeigeSteuerung2.end_Number;
            float newX = Mathf.Lerp(xleft, xright, t);
            // Update the x-position of the Pfeil
            pfeilTransform.localPosition = new Vector3(newX, pfeilTransform.localPosition.y, pfeilTransform.localPosition.z);
        }
    }
}